pathd/path_pcep.c

```diff
@@ -476,22 +479,35 @@ DEFUN(pcep_cli_pce, pcep_cli_pce_cmd,
 			i++;
 			continue;
 		}
+		if (strcmp("precedence", argv[i]->arg) == 0) {
+			i++;
+			if (i >= argc)
+				return CMD_ERR_NO_MATCH;
+			pce_precedence = atoi(argv[i]->arg);
+			i++;
+			continue;
+		}
 		return CMD_ERR_NO_MATCH;
 	}
 
 	pce_opts = XCALLOC(MTYPE_PCEP, sizeof(*pce_opts));
 	IPADDR_COPY(&pce_opts->addr, &pce_addr);
 	pce_opts->port = pce_port;
 	pce_opts->draft07 = draft07;
+	pce_opts->precedence = pce_precedence;
```

Comment:
These should not be stored in pathd/path_pcep.c pcep_glob.pce_opts[] since that is where the PCEs that are created with the new pcc-peer command will be stored. (Multiple) PCEs that are to be connected with the PCC should be stored in a new array in path_pcep_pcc.h in the pcc_state struct as struct pce_opts *pce_opts[MAX_PCE]; (basically change struct pce_opts *pce_opts to struct pce_opts *pce_opts[MAX_PCE])

Reply:
pcep_g level is for show command purposes.
For the name pce... vs pcc... maybe is has more sense but right now is some kind of pcc sessions